website/docs/deployment.mdx

```diff
@@ -140,14 +140,172 @@ cmd /C 'set "GIT_USER=<GITHUB_USERNAME>" && yarn deploy'
 
 [GitHub Actions](https://help.github.com/en/actions) allow you to automate, customize, and execute your software development workflows right in your repository.
 
-This workflow assumes your documentation resided in `documentation` branch of your repository and your [publishing source](https://help.github.com/en/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site) is configured for `gh-pages` branch.
+This workflow assumes your documentation resides in the `documentation` branch of your repository and your [publishing source](https://help.github.com/en/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site) is configured for the `gh-pages` branch.
+
+Below are two approaches to deploying your docs with GitHub Actions. Follow the example based on if your deployment branch (eg `gh-pages`) is in:
+
+- The **`same`** repository as your documentation source.
+- A **`remote`** repository different from your documentation source.
+
+````mdx-code-block
+<Tabs
+  defaultValue="same"
+  values={[
+    { label: 'Same', value: 'same' },
+    { label: 'Remote', value: 'remote' }
+]}>
+<TabItem value="same">
+
+```yaml title=".github/workflows/documentation-push.yml"
+# Appears in various parts of the Github UI usually with the job name appended.
+# (eg Project 'Actions' tab, PR check suite status)
+name: 'Docs (Github Pages)'
+
+on:
+  # Allows you to manually trigger the workflow for a commit
+  # (based on this workflow contents at that commit):
+  workflow_dispatch:
```

Comment:
I'm not actually sure I understood this event trigger correctly, and I think I should remove it.

Someone I was collaborating with suggested it for our projects workflow as it allows to manually trigger a workflow run instead of relying on a `push` event. This isn't on the `pull_request` event which got me curious. It seems to be for this purpose communicated to me, but you're meant to provide context for input parameters like a target commit or tag I think.

I'd need to investigate this further to confirm, but presumably without, you can still "re-run" any failed jobs which is useful if a failure was for temporary/external reasons such as network related interactions, or a conflict with pushing commits at deployment.

As I removed the workflow file path trigger, you might want to run the workflow when it's updated but nothing else is pushed or tagged to trigger the updated workflow to run. Using "re-run" would run the workflow from the state it was in at the commit it was run on, so a workflow "re-run" isn't always what a user wants to do.

Reply:
Yep, this trigger adds a manual trigger to the "Actions" tab of a project on Github.

Custom inputs can be provided but may require some additional handling to use those vs other triggers which will likely source those inputs from other sources (like env vars provided from the triggered event).

By default, it runs against the latest commit of any selected branch. I haven't setup a project where someone else who isn't the owner receives a PR from a fork, so I'm not quite sure how compatible it is with `pull_request` event if you ever wanted to use it under that context... but a "re-run" is probably sufficient for PRs which doesn't require this event trigger.

I'll keep it for `push` example, but update the comment for it.